### PR TITLE
Updated error parsers and exported more stuff

### PR DESCRIPTION
**Pull request by T-Damer**

### CODE REVIEW

File: src/helpers/parseError.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    parseRevertReason(displayedError) ||
+    parseCannotEstimateGasError(displayedError)
+  if (parsedError) displayedError = parsedError
+
+  // 4. Try to serialize it to RPC error
+  const serializedMessage = serializeError(displayedError || error, {
+    shouldIncludeStack: false,
+  }).message
+
+  return displayedError || serializedMessage || defaultMessage
```
Comment (T-Damer) [inline src/helpers/parseError.ts]: This defines the priority order

File: src/helpers/errorParsers.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+export function parseMetamaskReason(error: string) {
+  const metaMaskMessage = 'MetaMask Message Signature:'
+  if (error.includes(metaMaskMessage))
+    return error.split(metaMaskMessage)[1]?.split(' stack:')[0]
+}
+
+export function parseGSNError(error: string) {
+  return error.includes('Insufficient funds') ||
+    error.includes("reading 'error'")
```
Comment (T-Damer) [inline src/helpers/errorParsers.ts]: Not sure about this one, it appears when the mint is failed due to gsn
image

File: src/tests/exampleErrors.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  [errorName: string]: {
+    error: string | Record<string, unknown>
+    shouldDisplay: string
+  }
+}
+
+export const exampleErrors: ExampleErrors = {
+  unhandledRejection: {
+    error:
+      "Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'error')",
```
Comment (T-Damer) [inline src/tests/exampleErrors.ts]: Should this be unknown or gsn error?

File: vite.config.ts [hunk 4]
```diff
@@ -18,7 +18,7 @@ export default defineConfig({
       fileName: (format) => `index.${format}.js`,
     },
     rollupOptions: {
-      external: ['preact', 'react', 'react-toastify'],
+      external: ['preact', 'react', 'reactToastify', '/__tests__/'],
```
Comment (T-Damer) [inline vite.config.ts]: image
We had this warning, I fixed it

File: src/__tests__/parseErrors.test.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
+import { ErrorList } from '../types/ErrorList'
+import { expect, test } from '@jest/globals'
+import { parseError } from '../helpers/parseError'
+
+const exampleErrors = {
```
Comment (T-Damer) [inline src/__tests__/parseErrors.test.ts]: I can put it in another file outside `__tests__`

File: README.md [hunk 6]
```diff
@@ -11,6 +11,7 @@
 
 ## Available Scripts
 
-- `yarn start` — development mode
+- `yarn start` — development mode, rebuilds on changes
 - `yarn build` — builds utils
 - `yarn release` — create a release and publish the package
+- `yarn test:errors` to test error parsers
```
Comment (upacyxou) [inline README.md]: I think we should rename it to just `test`

Comment (upacyxou) [reply README.md]: Also, is it work with GH Actions?

File: src/__tests__/parseErrors.test.ts (hunk 5 above)
Comment (upacyxou) [reply src/__tests__/parseErrors.test.ts]: Put it into `utils` or something

File: src/__tests__/parseErrors.test.ts [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    shouldDisplay: ErrorList.gsnError,
+  },
+  unhandledRejection: {
+    error:
+      "Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'error')",
+    shouldDisplay: ErrorList.gsnError,
+  },
+}
+
+test('Error parser test', () => {
```
Comment (upacyxou) [inline src/__tests__/parseErrors.test.ts:5]: add negative cases

**Review by upacyxou — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (smyzin) [reply README.md]: Agreed, let's leave it as `yarn test`, as you did in the package.json

File: src/helpers/exapleErrors.ts [hunk 8]
```diff
@@ -0,0 +1,54 @@
+import { ErrorList } from '../types/ErrorList'
+
+export const exampleErrors = {
```
Comment (smyzin) [inline src/helpers/exapleErrors.ts]: Maybe we should move it somewhere to the test deir? It's strange that error examples in helpers

**Review by smyzin — APPROVED** [review]
Lgtm

File: src/helpers/exapleErrors.ts (hunk 8 above)
Comment (T-Damer) [reply src/helpers/exapleErrors.ts]: The jest will try to test the file than

**Review by smyzin — APPROVED** [review]
Lgtm

**Review by upacyxou — APPROVED** [review]

**Pull request merged**